FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.Core/Jobs/AzureStorageJobFactory.cs

```diff
@@ -0,0 +1,164 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using Azure.Data.Tables;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.Core.DataFilter;
+using Microsoft.Health.Fhir.Synapse.Core.DataProcessor;
+using Microsoft.Health.Fhir.Synapse.Core.Jobs.Models;
+using Microsoft.Health.Fhir.Synapse.DataClient;
+using Microsoft.Health.Fhir.Synapse.DataWriter;
+using Microsoft.Health.Fhir.Synapse.SchemaManagement;
+using Microsoft.Health.Fhir.Synapse.SchemaManagement.Parquet;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Synapse.Core.Jobs
+{
+    /// <summary>
+    /// Factory to create different jobs.
+    /// </summary>
+    public class AzureStorageJobFactory : IJobFactory
+    {
+        private readonly JobSchedulerConfiguration _schedulerConfiguration;
+        private readonly FilterConfiguration _filterConfiguration;
+        private readonly IQueueClient _queueClient;
+        private readonly IFhirDataClient _dataClient;
+        private readonly IFhirDataWriter _dataWriter;
+        private readonly ITypeFilterParser _typeFilterParser;
+        private readonly IGroupMemberExtractor _groupMemberExtractor;
+        private readonly IColumnDataProcessor _parquetDataProcessor;
+        private readonly TableClient _metaDataTableClient;
+        private readonly IFhirSchemaManager<FhirParquetSchemaNode> _fhirSchemaManager;
+        private readonly ILoggerFactory _loggerFactory;
+        private readonly ILogger<AzureStorageJobFactory> _logger;
+
+        public AzureStorageJobFactory(
+            IQueueClient queueClient,
+            IFhirDataClient dataClient,
+            IFhirDataWriter dataWriter,
+            ITypeFilterParser typeFilterParser,
+            IGroupMemberExtractor groupMemberExtractor,
+            IColumnDataProcessor parquetDataProcessor,
+            IFhirSchemaManager<FhirParquetSchemaNode> fhirSchemaManager,
+            IAzureTableClientFactory azureTableClientFactory,
+            IOptions<JobSchedulerConfiguration> schedulerConfiguration,
+            IOptions<FilterConfiguration> filterConfiguration,
```

Comment:
We should have a Filter configuration store here similar as IFhirSchemaManager. It is hard for wrapper project to extarct the configuration.

Reply:
Thanks, I will create it.

Reply:
Updated.